lib/std/Build/Step/Compile.zig

```diff
@@ -711,13 +712,14 @@ fn runPkgConfig(compile: *Compile, lib_name: []const u8) !PkgConfigResult {
             }
         }
 
-        // Now try appending ".0".
+        // Prefixed "lib" or suffixed ".0".
         for (pkgs) |pkg| {
             if (std.ascii.indexOfIgnoreCase(pkg.name, lib_name)) |pos| {
-                if (pos != 0) continue;
-                if (mem.eql(u8, pkg.name[lib_name.len..], ".0")) {
-                    break :match pkg.name;
-                }
+                const prefix = pkg.name[0..pos];
+                const suffix = pkg.name[pos + lib_name.len ..];
+                if (prefix.len > 0 and !mem.eql(u8, prefix, "lib")) continue;
+                if (suffix.len > 0 and !mem.eql(u8, suffix, ".0")) continue;
+                break :match pkg.name;
```

Comment:
Wouldn't this check for a `lib` prefix **and** a `.0` suffix?

Reply:
Fwict `prefix.len` and `suffix.len` are `0` if they are not present. That means `.len > 0` is `false`, the `continue` is skipped.
That means both `lib` prefix and `.0` suffix are allowed, but neither are required.
(Don't know whether that behavior is correct/intentional.)

Reply:
checking over it again, you are correct
either check is only performed if either section has a non zero length

additionally I think it also accept case-insensitive matches so the prior check could be removed

Reply:
> additionally I think it also accept case-insensitive matches so the prior check could be removed

I left the prior case-insensitive check so that any matches *without* a prefix or suffix take precedent over ones that do. In practice, this might be a non-issue. On my machine, the only affected library is "udev" vs. "libudev", the former of which seems misconfigured (it doesn't output any flags).

Reply:
thats fair
I've gone ahead and looked into the difference between `udev.pc` and `libudev.pc`:
the libudev one is what you expect, a pkg-config file to make use of libudev
the udev file on the other is part of udev, not libudev, and is intended to be used to query the systems relevant udev path
you can check this for yourself by running `pkg-config --variable udev_dir udev`